include/range/v3/range_concepts.hpp

```diff
@@ -141,6 +142,25 @@ namespace ranges
                     ));
             };
 
+            struct ContiguousRange
+              : refines<RandomAccessRange>
+            {
+                template<typename Rng>
+                using data_reference_t = decltype(*ranges::data(std::declval<Rng&>()));
+
+                template<typename Rng>
+                using datum_t = meta::_t<std::remove_reference<data_reference_t<Rng>>>;
+
+                template<typename Rng>
+                auto requires_(Rng && rng) -> decltype(
+                    concepts::valid_expr(
+                        concepts::model_of<Same, InputRange::value_t<Rng>,
+                            meta::_t<std::remove_cv<datum_t<Rng>>>>(),
+                        concepts::model_of<CommonReference, InputRange::reference_t<Rng>,
+                            data_reference_t<Rng>>()
+                    ));
+            };
```

Comment:
This `ContiguousRange` concept isn't so much half-baked as it is intentionally weak. A proper Ranges TS `ContiguousRange` concept would be simpler, but needs `ContiguousIterator` which cannot be implemented non-intrusively for non-pointers. Since I suspect we'll eventually want to replace this concept with something different, I'm intentionally not  providing an opt-out so that it cannot be used to overload against other range concepts.

`STL1ContiguousRange` might be a better name.